src/basic/process-util.c

```diff
@@ -1623,6 +1624,47 @@ int get_process_threads(pid_t pid) {
         return n;
 }
 
+int clone_vm_vfork(int (*callback)(void *), int flags, void *userdata) {
+        static size_t stack_size = 0;
+        char *stack;
+        int r;
+
+        assert(callback);
+        assert(!FLAGS_SET(flags, CLONE_PARENT_SETTID));
+        assert(!FLAGS_SET(flags, CLONE_CHILD_SETTID));
+        assert(!FLAGS_SET(flags, CLONE_SETTLS));
+
+        if (stack_size == 0) {
+                struct rlimit rl;
+
+                if (getrlimit(RLIMIT_STACK, &rl) < 0)
+                        return log_debug_errno(errno, "Failed to get RLIMIT_STACK: %m");
+
+                stack_size = ROUND_UP(rl.rlim_cur, page_size());
```

Comment:
uhm, this is not part of this diff? Where is it coming from? The only change in that file is the posix_spawn helper?

Reply:
this was some queued review item from june 19th... you might have rebased since then a couple of times, where the issue went away. I didn't realize I still had some reviews pending that old

Reply:
ah right, might have been from before the switch to posix_spawn, when there was some manual stack shenanigans due to the direct clone2()